packages/@lwc/synthetic-shadow/src/polyfills/window-event-target/polyfill.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { windowRemoveEventListener, windowAddEventListener } from '../../env/window';
+
+import { defineProperties, isUndefined, isTrue } from '@lwc/shared';
+import { eventTargetGetter } from '../../env/dom';
+import { isNodeShadowed } from '../../shared/node-ownership';
+
+// this method returns true if an event can be seen by a particular eventTarget,
+// otherwise it returns false, which means the listener will never be invoked.
+function isQualifyingEventTarget(currentTarget: Window, evt: Event): boolean {
+    const { composed } = evt;
+    const originalTarget = eventTargetGetter.call(evt);
+    if (originalTarget === currentTarget) {
+        // dispatched on the window directly
+        return true;
+    }
+    // if the event reaches the window by propagation is because it is bubbling,
+    // in which the the only it really depends on the composed value
+    return isTrue(composed) || !isNodeShadowed(originalTarget as Node);
```

Comment:
Is this safe for the case where we have third party libraries listening for events dispatched from manually inserted elements inside LWC?

Reply:
I want to flag this as a concern too. This is potentially blocking some non-composed events bubbling from an aura components within an lwc container from reaching the window. 
The [current behavior](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/synthetic-shadow/src/polyfills/event-listener/polyfill.ts) is only deciding whether to patch or not to patch an event reaching the window. It never really blocks the event from reaching the window.